lib/spack/docs/build_settings.rst

```diff
@@ -332,7 +332,7 @@ keyed by package name:
 
    packages:
      libfabric:
-       require: "@1.13.2"
+       require: ["@1.13.2"]
```

Comment:
It's not always very convenient because in some cases Spack tells you that the config file has to be updated for Spack to modify it. I think it's better to show this format in the docs, so that that doesn't happen, but also so that users do not get the impression that there's a generic rule `merge("abc", ["def"]) -> ["abc", "def"]`.